subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/plugins/PrecompiledGroovyPluginsIntegrationTest.groovy

```diff
@@ -766,7 +766,7 @@ class PrecompiledGroovyPluginsPluginIntegrationTest extends AbstractIntegrationS
     private void enablePrecompiledPluginsInBuildSrc() {
         file("buildSrc/build.gradle") << """
             plugins {
-                id 'precompiled-groovy-plugin'
+                id 'groovy-gradle-plugin'
```

Comment:
❓ We should maybe consider making the ID of plugins uniform between Groovy and Kotlin support.

cc @bamboo

Reply:
I totally agree, it's a great source of confusion. Also the Kotlin DSL plugin doesn't behave like a normal plugin which adds to the confusion. However, I think it can be done in a following PR.

Reply:
Yes, just to clarify I didn't mean to do anything as part of this PR.

Reply:
@eskatos I think so too.

> Also the Kotlin DSL plugin doesn't behave like a normal plugin which adds to the confusion. 

@lacasseio care to expand on that?

Reply:
I think it would be good to have a `groovy-gradle-plugin`, but I don't think it should be the name of the plugin that adds precompiled script plugin support.  So I would probably keep the "precompiled script" id for now.  A future groovy-gradle-plugin could apply this precompiled script plugin too (sort of like kotlin-dsl).

Reply:
Ok, before I change it again, let's decide what the name should be.

- `precompiled-groovy-gradle-plugin`
- `precompiled-groovy-plugin`
- something else?

Reply:
@bamboo It's not obvious at first that you need the Kotlin DSL plugin to properly write Gradle plugins. You can succeed up to a certain point with the normal Kotlin JVM plugin then start hitting errors when using it within the Kotlin DSL. The plugin is also treated specially when applying it within a plugin. You can have a plugin do something like `project.pluginManager.apply("kotlin-dsl")` like you would with any other plugin. You have to apply the plugin using the fully qualified name where you need to add a dependency to the binary from your plugin but because the version of that plugin is highly tied to the Gradle version, it's not possible to be future or backward compatible. There is a bunch of separated issues in there not related specifically to `kotlin-dsl` plugin however, it's the only core plugin that has a special treatment like that. It may have changed since the last time I looked into it which was around August.